### ISSUE TITLE
Different delegate methods for each Credentials plugin

### ISSUE THREAD

**Issue by crspybits** [IBM-Swift/Kitura-Credentials#42]
For the Google credentials, the CredentialsGoogleOptions.userProfileDelegate is currently defined as 

    public static let userProfileDelegate = "userProfileDelegate"

However, since the options seems scoped across all plugins (i.e., in the options property of a Credentials object), could CredentialsGoogleOptions.userProfileDelegate  be defined as:

 public static let userProfileDelegate = "googleUserProfileDelegate" ?

**Comment by irar2** [comment]
These are not the same options. There are two kinds of options: 

- Options passed to a plugin constructor. The keys are CredentialsGoogleOptions in the case of CredentialsGoogle plugin. 
- Options passed from Credentials to plugins' authenticate() methods during authentication process. These options are passed to Credentials.init(). CredentialsGoogle plugin ignores them at the moment.

**Comment by crspybits** [comment]
Thanks for your response. I'm not using the CredentialsGoogle plugin, I'm using the CredentialsGoogleToken plugin. I do not see an options parameter for that constructor. It seems that CredentialsGoogleToken takes its options only from the Credentials object.

**Comment by irar2** [comment]
You are right, but I think a better way to fix this is to add options to CredentialsGoogleToken constructor similar to CredentialsGoogle.

**Comment by crspybits** [comment]
👍

**Comment by irar2** [comment]
@crspybits Could you please verify that the problem is solved now?
